### ISSUE TITLE
TimeSeriesSVC's predict and predict_proba provide inconsistent results

### ISSUE THREAD

**Issue by rtavenar** [tslearn-team/tslearn#259]
**Describe the bug**
For class `TimeSeriesSVC`, `predict` and `predict_proba` provide inconsistent results.

**To Reproduce**
```python
from tslearn.svm import TimeSeriesSVC
import numpy as np


np.random.seed(0)
X = np.random.randn(6, 2, 1)
y = np.array([1, 1, 2, 2, 0, 0])

clf = TimeSeriesSVC(probability=True).fit(X, y)
print(clf.predict(X))
print(clf.predict_proba(X).argmax(axis=1))
```

gives

```
[1 1 2 2 0 0]
[0 2 0 0 1 2]
```

**Expected behavior**
`predict_proba` and `predict` should return consistent results, or the difference should be documented. 
Probably the best way for us at the moment would be to be consistent with sklearn (cf below).

**Additional info**
This follows a discussion in #256 and in fact the issue was already mentioned by @rth there: https://github.com/scikit-learn/scikit-learn/issues/13211

It seems allowing `predict_proba` (hence setting `probability=True`, which we do) for SVC has several pitfalls:
* makes the whole training process longer
* can lead to discrepancies between `predict` and `predict_proba` because of the way probabilities are computed
* can lead to issues with class orderings

**Suggestion for a fix**
At the moment, I think the following fix should be reasonable:
* have a `probability` parameter for our estimators as for sklearn ones (would make the training process faster for those who do not use probabilities)
  * if this is set to `False`, then `predict_proba` and `predict_log_proba` would not be accessible
  * if it is set to `True`, document the fact that it may provide inconsistent results (in the doc of the probabilities parameter and in that of the corresponding methods) and link to sklearn docs for more details.

**Comment by rtavenar** [comment]
After a quick check, in fact, we already have the `probability` attribute (and its default value is not properly documented, by the way!) that is transferred to sklearn model. 

So maybe it would just be a matter of documenting things better (fix the wrong default + have a BOLD WARNING) and skipping consistency checks for `TimeSeriesSVC`.

**Comment by GillesVandewiele** [comment]
In this specific case, the predictions are even way off (i.e. there's no possible mapping from the labels obtained `predict` and those obtained by `predict_proba`) due to the fact that predict_proba  of the SVC [uses 5-fold cross-validation internally](https://scikit-learn.org/dev/modules/generated/sklearn.svm.SVC.html#sklearn.svm.SVC). Perhaps this is something that could be handled by us as well? If the provided dataset is too small, one should probably fall back to using decision_function instead of predict_proba

**Comment by rtavenar** [comment]
I'd prefer we do not implement our own patch, but we could maybe have a gallery example or user guide section that would document the use of softmax (or sigmoid in the binary case) on top of `decision_function` for people willing to have consistent probabilities.

**Comment by rtavenar** [comment]
Maybe I'll try to have a user guide on kernel methods in #256 in which I could discuss that + fix the docs for the default value of `probability`.
